src/HtmlProcessor/AbstractHtmlProcessor.php

```diff
@@ -103,12 +103,40 @@ private function setHtml(string $html)
      * Provides access to the internal DOMDocument representation of the HTML in its current state.
      *
      * @return \DOMDocument
+     *
+     * @throws \UnexpectedValueException
      */
     public function getDomDocument(): \DOMDocument
     {
+        if (null === $this->domDocument) {
+            throw new \UnexpectedValueException(
+                self::class .
+                '::setDomDocument() has not yet been called on ' .
+                static::class
+            );
+        }
+
         return $this->domDocument;
     }
 
+    /**
+     * Provides access to the the internal DOMXPath instance corresponding to AbstractHtmlProcessor::$domDocument
```

Comment:
The intent of this comment was to highlight the relationship between the `DOMXpath` instance and the `DOMDocument`, can reword at your preference though.

Reply:
Yes, it's true that the `DOMXPath` is tied to the `DOMDocument`, however the specific class property that the method accesses is `xPath` rather than `domDocument`.

Reply:
leave this as pending for resolution of #789 ?

Reply:
Looking again, I think this actually is fine as it is.  I had initially thought it was the result of an oversight copy-and-pasting from `getDomDocument()`.